### ISSUE TITLE
search_fullarchive issues multiple requests for one individual request

### ISSUE THREAD

**Issue by danweitzel** [ropensci/rtweet#347]
### Problem

I am trying to use the search_fullarchive function and a 500 request premium account on Twitter to collect 500 tweets from 485 accounts in a specific 30 day period. I run a request for each individual account and expected that collecting 500 tweets from one account in a 31 day period would count as one request. However, the Twitter dashboard shows many more requests than actual accounts collected. I average 2.35 requests for one account (213 accounts collected with 500 requests). 

### Expected behavior

I expected that collecting 500 tweets from one account for a 31 day period would result in one request. 

### Reproduce the problem
``` r
for (i in 1:length(usernames)){
  tweets <- lapply(usernames[i], function(x) search_fullarchive(x, 
                                                                   n = 500, 
                                                                   env_name = "election15",
                                                                   fromDate = "201504080000", 
                                                                   toDate = "201505082359"))
  csvfile <- paste0("tweets_",i,".csv")
  tweets <- do.call(rbind.data.frame, tweets)
  tweets <- data.frame(lapply(tweets, as.character), stringsAsFactors=FALSE)
  write.csv(tweets, csvfile, row.names=FALSE)
  print(paste("I collected tweets from", usernames[i]))
  Sys.sleep(10)
}
```

### rtweet version
``` r
## copy/paste output
packageVersion("rtweet")
‘0.6.9’
```

**Comment by DonBunk** [comment]
I am not sure, but I suspect there is a maximum chunk for a given request. I cannot find it in the documentation for the full archive search, but the corresponding max chunk for the standard search is under the count parameter at https://developer.twitter.com/en/docs/tweets/search/api-reference/get-search-tweets.html. For the standard search, the chunk size is 100. If I understand your math correctly you are using ~2.5 requests for 500 tweets, which seems to be in line with a chunk size of ~100 as well.

**Comment by danweitzel** [comment]
You are correct! I checked the Twitter help file for premium search and the rtweet code on GitHub. 
Twitter's fullarchive search has a maxResults parameter that can range from 10-500 and defaults to 100: (https://developer.twitter.com/en/docs/tweets/search/api-reference/premium-search). 100 is also the maximum value the request can take in the Sandbox environment. 

I went through the `search_tweets.R` script on [GitHub](https://github.com/mkearney/rtweet/blob/master/R/search_tweets.R) and found the following code snippet that seems to indicate that the fullarchive search in rtweet is set to 100. 

```
if (grepl("fullarchive|30day", query)) {
    params[["premium"]] <- NULL
    params$result_type <- NULL
    if (grepl("full", query)) {
      params$maxResults <- 100
    } else {
      params$maxResults <- 100
    }
```
I also searched for documentation on the maxResults parameter in the rtweet help files. Am I correct in assuming that this is not a parameter that I can change?

**Comment by DonBunk** [comment]
I am not sure, but I gather 100 is the max. I haven't been able to find any way around the corresponding max values for the requests I have been making. 

Also, I don't know if it helps or not, but I often do small searches first and then use the rate_limit() function to calibrate my expectations for larger searches. Note that function is a bit buggy(rtweet or maybe Twitter itself) so sometimes I find it takes a few refreshes to accurately display the data.

**Comment by danweitzel** [comment]
It is an issue other's have raised, too. Issue #317 is about the same problem and @sentenza3000 is encountering the same hard-coded 100 `maxResult` limit for the fullarchive search that I am struggling with. 

I don't know if it would be possible to have the `maxResult` parameter either default to the maximum parameter that is available for a given search or be set by the users just like `n`, `fromDate`, and `toDate`. It is also entirely possible that I am missing something very obvious and the parameter can be set.

**Comment by DonBunk** [comment]
You probably already thought about this, but you could always clone the repository, set it to 101 and see what happens. You could also just just try making requests more directly via twurl (https://github.com/twitter/twurl) to see if a larger number works there.

**Comment by kevintaylor** [comment]
Just checking in on this. Has anyone found a reasonable solution since the last post in August 2019?

**Comment by kevintaylor** [comment]
@DonBunk, what rate limit do you query for when checking for full archive premium searches?

**Comment by kevintaylor** [comment]
I fixed this using @danweitzel and @DonBunk suggestions above by cloning and fixing the bug in search_tweets.R.

See here for code/explanation of search_fullarchive():

https://github.com/ropensci/rtweet/issues/368

**Comment by sentenza3000** [comment]
Awesome! I had not found a solution so far. This should also solve issue [https://github.com/ropensci/rtweet/issues/317#issuecomment-517364059](https://github.com/ropensci/rtweet/issues/317#issuecomment-517364059).
Do I understand correctly, that so far it is not implemented in the CRAN version of rtweet, but each user would still have to implement it individually?

**Comment by kevintaylor** [comment]
The fork with the fix is here, if someone needs to use the fix and doesn't want to clone and edit the search_tweets.R code themselves:

https://github.com/kevintaylor/rtweet

**Comment by norcalbiostat** [comment]
@kevintaylor - i've actually been having mixed results with your fork. I removed the original package and installed from github your fork, and confirmed that it was working (1 api call per 500 tweets)
But i just did the same for a different computer, and it's still using 1 call per 100. 
Where can I find the `get_tweets.R` code file on my local installation so I can adjust that function manually?

**Comment by kevintaylor** [comment]
@norcalbiostat, it may be that you have more than one version of rtweet installed on your system. I use RStudio and I can see both versions in my "packages" pane. I make sure my local version is the one that is checked. Where you can find the packages on your system depends on a few things. Here is one guide: https://www.r-bloggers.com/package-paths-in-r/

**Comment by norcalbiostat** [comment]
@kevintaylor Nope. I made sure I cleared out all instances of that package and only installed your fork. It's still using 5 calls per request. I checked all locations as well.

**Comment by dfornis** [comment]
@norcalbiostat you can do an edit with
edit(getAnywhere(.search_tweets))
You'll find the code there. Not sure if it saves for the next R session though

**Comment by dfornis** [comment]
I have an issue of my own:
search_fullarchive retrieves more number of tweets than specified by n =
It just seems to max out and retrieve all tweets relevant for my query. This consumes all my requests in no time.
search_30day works fine! Strangely enough.
This is an example of my code:
[
![Skärmavbild 2020-04-21 kl  12 32 58](https://user-images.githubusercontent.com/64075403/79856463-adc29900-83cc-11ea-8dc1-b9dafaf62d04.png)
](url)

**Comment by gabrielherrera27** [comment]
I had the same issue @dfornis when using search_fullarchive
I did some tests and it seems it goes like this:

set n=50 -- you get 500
set n=150 -- you get 1000
set n=250 -- you get 1500
set n=350 -- you get 2000 ... and so on

But it comes in blocks of 500 and burns less requests, so no big deal! Thanks @kevintaylor !

**Comment by llrs** [comment]
Closing as duplicate of #317, and #368. That is fixed once #375 gets merged.
